src/menu.js

```diff
@@ -458,13 +459,32 @@ exports.get = function(props = {}) {
                 .map(syncer => syncer.id)[0]
 
             if (syncerId == null) {
-              dialog.showMessageBox(
-                i18n.t(
-                  'menu.engines',
-                  'None of the attached engines support analysis.'
-                ),
-                'info'
-              )
+              let defaultEngine = setting
+                .get('engines.list')
+                .filter(engine => engine.analysis)[0]
+
+              if (defaultEngine != null) {
+                if (!analyzeEngineInitializing) {
+                  analyzeEngineInitializing = true
```

Comment:
I don't like this global variable here in `menu.js`. We could consider blocking the whole UI with `sabaki.setBusy(true)` (then menu items will be disabled) and after the engine is initialized, call `sabaki.setBusy(false)`.

Reply:
Ok, I removed 'analyzeEngineInitializing' and use sabaki.state.busy and sabaki.setBusy() instead.